@here/olp-sdk-core/lib/utils/MultiPartUpload.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import {
+    DataStoreRequestBuilder,
+    HRN,
+    OlpClientSettings,
+    RequestFactory
+} from "@here/olp-sdk-core";
+import { BlobApi, ObjectStoreApi } from "@here/olp-sdk-dataservice-api";
+
+interface MultiPartUploadParams {
+    /**
+     * The instance of OlpClientSettings.
+     */
+    settings: OlpClientSettings;
+    /**
+     * The version od Blob API to upload.
+     */
+    blobVersion: "v1" | "v2";
+    /**
+     * The Layer ID to upload.
+     */
+    layerId: string;
+    /**
+     * If you're uploading to Blob V1, set the datahandle
+     * and set the key for Blob V2.
+     */
+    keyOrDatahandle: string;
+    /**
+     * The Catalog HRN to upload.
+     */
+    catalogHrn: HRN;
+    /**
+     * The content type of a uploaded data.
+     * @see https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.17
+     */
+    contentType: string;
+    /**
+     * The Content Encoding of the data
+     * @note Optional
+     */
+    contentEncoding?: string;
+}
+
+interface UploadParams {
+    file: File | string;
+    chunkSize: number;
```

Comment:
This should be defined by the API or as a const internally. Is it really the users responsibility to provide this? What if he specifies a bigger value than allowed by the REST API?

Reply:
I've added comment about it to the method "upload".

```
* If you need to upload a large file in Browser, set the
     * File instance as a `file` parameter.
     * @see https://developer.mozilla.org/en-US/docs/Web/API/File
     *
     * For NodeJS - set the path to the file as a `file` parameter.
 ```

Means, if user use browser - should pass File object (we'll read that file chunk by chunk), if NodeJS - pass the path to the file (we'll read that file as a stream)